doc/extdev/index.rst

```diff
@@ -189,6 +189,123 @@ as metadata of the extension.  Metadata keys currently recognized are:
   usually don't negatively influence the process, this defaults to ``True``.
 
 
+Extension How-tos
+-----------------
+
+This section contains several how-tos associated with developing extensions in
+Sphinx.
+
+Defining custom template functions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Sometimes it is useful to define your own function in Python that you wish to
+then use in a template. For example, if you'd like to insert a template value
+with logic that depends on the user's configuration in the site, or if you'd
+like to include non-trivial checks and provide friendly error messages for
+incorrect configuration in the template.
+
+To define your own template function, you'll need to define two functions inside
+your module, a **parent function** and a **template function**.
+
+First, define the **parent function**, which accepts the
+arguments for :event:`html-page-context`.
+
+Within the parent function, define
```

Comment:
You don't have to define template functions inside event handler. I feel defining them on module level is simple and easy for test.

Reply:
Interesting - I assumed that this was necessary in order to have the template function get access to all of the variables that are passed to the event handler function (e.g., `app`, `context`, etc). Is that not the case?

Reply:
I just confirmed that (in another project), when I defined the function outside of the event handler function, then the template function no longer had access to the `context` variable

Reply:
Sure. You need to define it inside handler function if you need to access it. Okay, let's go as is.